api/http/handler/kubernetes/namespaces.go

```diff
@@ -1,43 +1,100 @@
 // @failure 400 "Invalid request"
 // @failure 500 "Server error"
 // @router /kubernetes/{id}/namespaces [get]
-func (handler *Handler) getKubernetesNamespaces(w http.ResponseWriter, r *http.Request) *httperror.HandlerError {
-	cli, handlerErr := handler.getProxyKubeClient(r)
-	if handlerErr != nil {
-		return handlerErr
+func (handler *Handler) GetKubernetesNamespaces(w http.ResponseWriter, r *http.Request) *httperror.HandlerError {
+	namespaces, err := handler.getKubernetesNamespaces(w, r)
+	if err != nil {
+		return err
 	}
 
-	namespaces, err := cli.GetNamespaces()
+	return response.JSON(w, namespaces)
+}
+
+// @id GetKubernetesNamespacesCount
+// @summary Get the total number of kubernetes namespaces within the given Portainer environment.
+// @description Get the total number of kubernetes namespaces within the given environment (Endpoint), including the system namespaces. The total count depends on the user's role and permissions. The Endpoint ID must be a valid Portainer environment identifier.
+// @description **Access policy**: authenticated
+// @tags kubernetes
+// @security ApiKeyAuth
+// @security jwt
+// @accept json
+// @produce json
+// @param id path int true "Environment (Endpoint) identifier"
+// @success 200 {integer} integer "Success"
+// @failure 400 "Invalid request"
+// @failure 500 "Server error"
+// @router /kubernetes/{id}/namespaces/count [get]
+func (handler *Handler) getKubernetesNamespacesCount(w http.ResponseWriter, r *http.Request) *httperror.HandlerError {
+	namespaces, err := handler.getKubernetesNamespaces(w, r)
 	if err != nil {
-		return httperror.InternalServerError("Unable to retrieve namespaces", err)
+		return err
 	}
 
-	return response.JSON(w, namespaces)
+	return response.JSON(w, len(namespaces))
 }
 
-// @id getKubernetesNamespace
+func (handler *Handler) getKubernetesNamespaces(w http.ResponseWriter, r *http.Request) (map[string]portainer.K8sNamespaceInfo, *httperror.HandlerError) {
+	withResourceQuota, err := request.RetrieveBooleanQueryParameter(r, "withResourceQuota", false)
+	if err != nil {
+		return nil, httperror.BadRequest("an error occurred during the getKubernetesNamespaces operation, invalid query parameter withResourceQuota. Error: ", err)
+	}
+
+	cli, httpErr := handler.getProxyKubeClient(r)
+	if httpErr != nil {
+		return nil, httperror.InternalServerError("an error occurred during the getKubernetesNamespacesCount operation, unable to get a Kubernetes client for the user. Error: ", httpErr)
+	}
+
+	endpoint, err := middlewares.FetchEndpoint(r)
+	if err != nil {
+		return nil, httperror.NotFound("Unable to find an environment on request context", err)
+	}
+
+	pcli, err := handler.KubernetesClientFactory.GetPrivilegedKubeClient(endpoint)
+	if err != nil {
+		return nil, httperror.InternalServerError("an error occurred during the getKubernetesNamespaces operation, unable to get a privileged Kubernetes client for the user. Error: ", err)
+	}
+	pcli.IsKubeAdmin = cli.IsKubeAdmin
```

Comment:
I'm not a huge fan of this pattern, the pcli object being a privileged Kubernetes client it should not have any property related to being an administrator or having namespace accesses - being privileged means it should be user agnostic.

Let's discuss this to see if we can find a better alternative but I think that you might want to build specific functions for a privileged Kubernetes client that take these parameters in their signature.